android/filament-utils-android/src/main/java/com/google/android/filament/utils/KtxLoader.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.filament.utils
+
+import com.google.android.filament.Engine
+import com.google.android.filament.IndirectLight
+import com.google.android.filament.Skybox
+import com.google.android.filament.Texture
+
+import java.nio.Buffer
+
+/**
+ * Utilities for consuming KTX files and producing Filament textures, IBLs, and sky boxes.
+ *
+ * KTX is a simple container format that makes it easy to bundle miplevels and cubemap faces
+ * into a single file.
+ */
+object KtxLoader {
+    class Options {
+        var srgb = false
+    }
+
+    /**
+     * Consumes the content of a KTX file and produces a [Texture] object.
+     *
+     * @param engine Gets passed to the builder.
+     * @param buffer The content of the KTX File.
+     * @param options Loader options.
+     * @return The resulting Filament texture, or null on failure.
+     */
+    fun createTexture(engine: Engine, buffer: Buffer, options: Options): Texture {
```

Comment:
Should the return type be `Texture?` since it can be `null`? This applies to `createIndirectLight` and `createSkybox` as well.

Reply:
I think the return type should actually stay non-nullable because it returns a Texture object that simply wraps a native texture.

Currently the native KtxLoader is not graceful if something goes wrong, it throws a NPE.  I should fix this up in a subsequent PR and make this nullable.

Reply:
Sounds good